commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/ext/XMLStreamReaderExt.kt

```diff
@@ -0,0 +1,18 @@
+package ac.uk.ebi.biostd.ext
+
+import javax.xml.stream.XMLStreamReader
+
+/**
+ * Iterates over each XML node loaded by the reader and applies the given function .
+ *
+ * @param function The function to be applied to each element.
+ */
+fun XMLStreamReader.forEach(function: () -> Unit) {
+    while (hasNext()) {
+        try {
+            function()
+        } catch (exception: NoSuchElementException) {
+            break
```

Comment:
we should use exceptional code "normal code", we need to identify when there is not more reader and finalize buckle is like making this.

```
var list = listOf(10, 22)
index = 0;
while(true) {
     try {
     print(list[index++])
    }
    catch (exception: NoSuchElementException)
        break;
} 
```